pkg/operation/botanist/controlplane.go

```diff
@@ -206,3 +212,14 @@ func (b *Botanist) RestartControlPlanePods(ctx context.Context) error {
 		client.MatchingLabels{v1beta1constants.LabelPodMaintenanceRestart: "true"},
 	)
 }
+
+func waitUntilPodsAreTerminatedForDeployments(ctx context.Context, c client.Client, namespace string, deployments []string) error {
```

Comment:
```suggestion
func waitUntilNoPodsExistAnymore(ctx context.Context, c client.Client, namespace string, deploymentNames ...string) error {
```

Reply:
I renamed the function, but left the last parameter as slice since this is not exported. We can always change it later if there is the use case for it.